lib/srv/app/aws/handler.go

```diff
@@ -145,46 +142,27 @@ func (s *SigningService) RoundTrip(req *http.Request) (*http.Response, error) {
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	signedReq, err := s.prepareSignedRequest(req, resolvedEndpoint, sessionCtx)
+	payload, err := awsutils.GetAndReplaceReqBody(req)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	resp, err := s.Client.Do(signedReq)
+	signedReq, err := s.prepareSignedRequest(req, payload, resolvedEndpoint, sessionCtx)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-
-	if err := s.emitAuditEvent(req.Context(), signedReq, resp, sessionCtx, resolvedEndpoint); err != nil {
+	resp, err := s.RoundTripper.RoundTrip(signedReq)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	// restore signedReq body since the transport closed it.
+	signedReq.Body = io.NopCloser(bytes.NewReader(payload))
+	err = sessionCtx.Audit.OnRequest(req.Context(), sessionCtx, signedReq, resp, resolvedEndpoint)
```

Comment:
Actually I changed this and it broke tests because the signedReq is the one with the resolved endpoint url and host. We don't extract any SigV4 AWS headers.